virtcontainers/pkg/oci/utils.go

```diff
@@ -342,14 +355,65 @@ func addAssetAnnotations(ocispec specs.Spec, config *vc.SandboxConfig) {
 
 		config.Annotations[a] = value
 	}
+}
+
+func addHypervisorConfigOverrides(ocispec specs.Spec, config *vc.SandboxConfig) error {
+
+	if value, ok := ocispec.Annotations[vcAnnotations.DefaultVCPUs]; ok {
+		vcpus, err := strconv.ParseUint(value, 10, 32)
+		if err != nil {
+			return fmt.Errorf("Error encountered parsing annotation for default_vcpus: %v, please specify numeric value", err)
+		}
+
+		numCPUs := goruntime.NumCPU()
+
+		// Assign the override only if value is valid, else fallback to what is provided by the config
+		if uint32(vcpus) < uint32(numCPUs) {
+			config.HypervisorConfig.NumVCPUs = uint32(vcpus)
```

Comment:
A warning may be helpful here in logs

Reply:
Or just fail is good as well

Reply:
Yeah, I was thinking what the behaviour should be here while implementing this, was leaning towards just erroring out here myself. But ended up implementing behaviour similar to what we have for config toml file, where we fallback to defaults we have in the runtime.
I can certainly error out here. We should probably change how we handle configuration from the config.toml file to keep things consistent.
Another thing to consider maybe we could just assign the max number of CPUs on the system when the number requested exceeds that, especially if you have a heterogeneous cluster.
cc @jcvenegas @egernst